plugins/scheduler-docker-local/check-deploy

```diff
@@ -176,48 +195,20 @@ scheduler-docker-local-check-deploy() {
       local CURL_OPTIONS+=" -H X-Forwarded-Proto:https"
     fi
 
-    exec <"$CHECKS_FILENAME"
-    local CHECK_URL
-    local EXPECTED
-    while read -r CHECK_URL EXPECTED; do
-      # Ignore empty lines and lines starting with #
-      # shellcheck disable=SC1001
-      [[ -z "$CHECK_URL" || "$CHECK_URL" =~ ^\# ]] && continue
-      # Ignore if it's not a URL in a supported format
-      # shellcheck disable=SC1001
-      ! [[ "$CHECK_URL" =~ ^(http(s)?:)?\/.* ]] && continue
-
-      if [[ "$CHECK_URL" =~ ^https?: ]]; then
-        local URL_PROTOCOL=${CHECK_URL%:*}
-        local CHECK_URL=${CHECK_URL#*:}
-      else
-        local URL_PROTOCOL="http"
-      fi
+    # This URL will show up in the messages
+    local LOG_URL="$URL_PROTOCOL://$URL_HOSTNAME$URL_PATHNAME"
+    # And how we formulate the CURL request
+    local CURL_ARGS="$CURL_OPTIONS $URL_PROTOCOL://$DOKKU_APP_LISTEN_IP:$DOKKU_APP_LISTEN_PORT$URL_PATHNAME $HEADERS"
 
-      if [[ "$CHECK_URL" =~ ^//.+ ]]; then
-        # To test a URL with specific host name, we still make request to localhost,
-        # but we set Host header to $SEND_HOST.
-        #
-        # The pattern is
-        #   //SEND_HOST/PATHNAME
-        local UNPREFIXED=${CHECK_URL#//}
-        local URL_HOSTNAME=${UNPREFIXED%%/*}
-        local URL_PATHNAME=${UNPREFIXED#$URL_HOSTNAME}
-
-        local HEADERS="-H Host:$URL_HOSTNAME"
-      else
-        local URL_HOSTNAME=localhost
-        local URL_PATHNAME=$CHECK_URL
-      fi
-
-      # This URL will show up in the messages
-      local LOG_URL="$URL_PROTOCOL://$URL_HOSTNAME$URL_PATHNAME"
-      # And how we formulate the CURL request
-      local CURL_ARGS="$CURL_OPTIONS $URL_PROTOCOL://$DOKKU_APP_LISTEN_IP:$DOKKU_APP_LISTEN_PORT$URL_PATHNAME $HEADERS"
+    dokku_log_verbose "CHECKS expected result:"
+    dokku_log_verbose "$LOG_URL => \"$EXPECTED\""
+    [[ $DOKKU_TRACE ]] && dokku_log_verbose "$ curl $CURL_ARGS"
 
-      dokku_log_verbose "CHECKS expected result:"
-      dokku_log_verbose "$LOG_URL => \"$EXPECTED\""
-      [[ $DOKKU_TRACE ]] && dokku_log_verbose "$ curl $CURL_ARGS"
+    local ATTEMPT=0
+    until [[ $SUCCESS == 1 || $ATTEMPT -ge $ATTEMPTS ]]; do
+      local ATTEMPT=$((ATTEMPT + 1))
+      dokku_log_verbose "Attempt $ATTEMPT/$ATTEMPTS. Waiting for $WAIT seconds ..."
```

Comment:
again i think it's important to communicate when we are intentionally not proceeding